plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitSource.java

```diff
@@ -69,6 +71,13 @@
 {
     private static final Logger log = Logger.get(HiveSplitSource.class);
 
+    private static final int PER_SPLIT_ROW_ID_BITS = 42;
+    private static final int SPLIT_ID_BITS = 64 - PER_SPLIT_ROW_ID_BITS;
+    // We partitions the rowId space between splits assigning each split 2^42 ids.
+    // As we need to encode the split number in id it allows us to have at most 2^22 splits per query
```

Comment:
nit: move comment to the top of section which devices 4 constants

Reply:
sure